pyogrio/_io.pyx

```diff
@@ -1678,6 +1678,117 @@ cdef infer_field_types(list dtypes):
 
     return field_types
 
+def ogr_write_arrow(
+    obj,
+    str path
+):
+    IF CTE_GDAL_VERSION < (3, 8, 0):
+        raise RuntimeError("Need GDAL>=3.8 for Arrow write support")
+
+    stream_capsule = obj.__arrow_c_stream__()
+    return write_arrow_stream_capsule(stream_capsule)
+
+cdef OGRErr write_arrow_stream_capsule(OGRLayerH destLayer, object capsule):
+    cdef ArrowArrayStream stream
+    cdef ArrowSchema schema
+    cdef ArrowArray array
```

Comment:
```suggestion
    cdef ArrowArrayStream* stream
    cdef ArrowSchema* schema
    cdef ArrowArray* array
```

(I would expect we are working with pointers here, but didn't look in detail at how those variables are used in the code)

Reply:
Or at least for the stream, maybe not for the schema/array

Reply:
Yes, but there we allocate those structs ourselves (to let them be filled in by GDAl and to export them), while here we get it from the capsule

Reply:
😅 I can't write anything without the Rust IDE telling me exactly what to do